extensions/sql-migration/src/wizard/wizardController.ts

```diff
@@ -81,9 +84,44 @@ export class WizardController {
 		await Promise.all(wizardSetupPromises);
 		await pages[0].onPageEnter();
 
-		this.extensionContext.subscriptions.push(wizard.doneButton.onClick(async (e) => {
+		this.extensionContext.subscriptions.push(this._wizardObject.doneButton.onClick(async (e) => {
 			await stateModel.startMigration();
 		}));
+
+		this._wizardObject.cancelButton.onClick(e => {
+			sendSqlMigrationActionEvent(
+				TelemetryViews.SqlMigrationWizard,
+				TelemetryAction.PageButtonAction,
+				{
+					'sessionId': this._model._sessionId,
+					'buttonPressed': 'cancel',
+					'pageTitle': this._wizardObject.pages[this._wizardObject.currentPage].title
+				}, {});
+		});
+
+		this._wizardObject.doneButton.onClick(e => {
+			sendSqlMigrationActionEvent(
+				TelemetryViews.SqlMigrationWizard,
+				TelemetryAction.PageButtonAction,
+				{
+					'sessionId': this._model._sessionId,
+					'buttonPressed': 'done',
+					'pageTitle': this._wizardObject.pages[this._wizardObject.currentPage].title
+				}, {});
+		});
+	}
+
+	private async generateWizardTelemetry(pageChangeInfo: azdata.window.WizardPageChangeInfo) {
```

Comment:
Does this need to be async?

Reply:
I wanted to make it non-blocking.

Reply:
But you're not currently awaiting anything so async is unnecessary in this case